conans/client/tools/win.py

```diff
@@ -451,15 +463,14 @@ def run_in_windows_bash(conanfile, bashcmd, cwd=None, subsystem=None, msys_mingw
             # Put the build_requires and requires path at the first place inside the shell
             hack_env = ' && PATH="%s:$PATH"' % inherited_path if inherited_path else ""
 
-        for var_name, value in env.items():
-            if var_name == "PATH":
-                continue
-            hack_env += ' && %s=%s' % (var_name, value)
+            for var_name, value in env.items():
+                if var_name == "PATH":
+                    continue
+                hack_env += ' && %s=%s' % (var_name, value)
```

Comment:
Are you missing quotes around value?
hack_env += ' && %s="%s"' % (var_name, value)

Reply:
It's also missing an export for processes started from the shell 
```hack_env += ' && export %s="%s"' % (var_name, value)```